src/Tickets/Commerce/Gateways/Contracts/Abstract_Gateway.php

```diff
@@ -302,4 +311,25 @@ public static function disable() {
 
 		return tribe_remove_option( static::get_enabled_option_key() );
 	}
+
+	/**
+	 * Is currency supported.
+	 *
+	 * @since TBD
+	 * 
+	 * @param string $currency_code Currency code.
+	 *
+	 * @return bool
+	 */
+	public static function is_currency_supported( $currency_code ) {
+		if ( empty( $currency_code ) ) {
+			return false;
+		}
+		
+		// If supported currencies aren't set, assume it's supported.
+		if ( empty( static::$supported_currencies ) ) {
```

Comment:
As a possible improvement for the future. I'd think of having a getter of supported currencies that return the value of the static $supported_currencies, filtered. So that if we find ourselves in the position in which a currency is no longer supported, or a gateway adds support for a new currency we don't have to wait until we fix that in a new release, and we can give a snippet to customers.

Reply:
Good idea. I'll add that.